packages/ui/src/Button.tsx

```diff
@@ -45,28 +45,47 @@ export type ButtonAccessibleIconRightProps =
       iconRightClassName?: never
     }
 
-export type ButtonDisabledProps =
-  | { disabledTooltip: string; disabled: boolean; disabledTooltipVisible?: boolean; disabledTooltipPlacement?: TooltipProps['placement'] }
-  | {
-      disabled?: never
-      disabledTooltip?: never
-      disabledTooltipVisible?: never
-      disabledTooltipPlacement?: never
-    }
+type ButtonNotDisabledProps = {
+  disabled?: never
+  disabledTooltip?: never
+  disabledTooltipVisible?: never
+  disabledTooltipPlacement?: never
+}
+
+export type ButtonDisabledProps = {
+  disabledTooltip: string
+  disabled: boolean
+  disabledTooltipVisible?: boolean
+  disabledTooltipPlacement?: TooltipProps['placement']
+}
 
 // Common props for all button "kinds"
 type ButtonCommonProps = {
   kind?: ButtonKind
   children: string
   capitalize?: boolean
   bodyClassName?: string
-}
+} & Pick<HTMLAttributes<HTMLAnchorElement | HTMLButtonElement>, 'className' | 'onClick'>
+
+type ButtonTypeProps =
+  | ({
+      component: 'a'
+      href: string
+    } & ButtonNotDisabledProps)
+  | ((
+      | {
+          component: 'button'
+          href?: never
+        }
+      | {
+          component?: undefined
+          href?: never
+        }
+    ) &
+      (ButtonDisabledProps | ButtonNotDisabledProps) &
+      Pick<ButtonHTMLAttributes<HTMLButtonElement>, 'autoFocus' | 'type'>)
```

Comment:
This can be simplified if I'm not mistaken:
```suggestion
type ButtonTypeProps =
  | ({
      component: 'a'
      href: string
    } & ButtonNotDisabledProps)
  | ({
      component?: 'button'
      href?: never
    } & (ButtonDisabledProps | ButtonNotDisabledProps) &
      Pick<ButtonHTMLAttributes<HTMLButtonElement>, 'autoFocus' | 'type'>)
```